src/main_dpdk.cpp

```diff
@@ -3568,9 +3588,76 @@ COLD_FUNC void CGlobalTRex::rx_interactive_conf(void) {
     }
 }
 
+/**
+ * Decides if `m` is a PTP packet that sould be timestamped (only event messages are interesting).
+ * If so, returns ptp header offset in the L2 frame.
+ */
+static inline HOT_FUNC uint8_t get_ptp_packet_offset(struct rte_mbuf *m) {
+    uint8_t *pkt = rte_pktmbuf_mtod(m, uint8_t *);
```

Comment:
it would be nice to decrease the number of operations in callback functions, but we can do this later if we have time

Reply:
I couldn't agree more :)

Reply:
The problem is however TRex + DPDK do not provide all the information we need to determine packet's content.  At least not always.  That is why we chose to "parse" those packets in the callbacks.

Reply:
I guess it might be the procedure of getting the timestamps (from the NIC) which takes longest.  So it is probably wise to do this "parsing" before actually calling `CGlobalInfo::m_options.get_latency_timestamp(port);`  On the other hand I might be wrong